src/infrastructure/content/server/ContentServer.java

```diff
@@ -0,0 +1,243 @@
+			logger.log(ModuleID.INFRASTRUCTURE, LogLevel.SUCCESS, logMessage);
+		}
+		else if (meta.equals("message")) {
+			// logging to logger 
+			logMessage = "Message case";
+			logger.log(ModuleID.INFRASTRUCTURE, LogLevel.INFO, logMessage);
+			// saving the local time
+			LocalTime now = LocalTime.now();
+			DateTimeFormatter dtf = DateTimeFormatter.ofPattern("HH:mm:ss");
+			try {
+				time = dtf.format(now).toString();
+			} catch(Exception e) {
+				// logging to logger
+				logMessage = "Error has occured. The current time cannot be formatted.";
+				logger.log(ModuleID.INFRASTRUCTURE, LogLevel.ERROR, logMessage);
+				return;
+			}
+			jsonObject.put("time",time);
+			// logging to logger
+			logMessage = "The message is now being broadcasted to all clients";
+			logger.log(ModuleID.INFRASTRUCTURE, LogLevel.INFO, logMessage);
+			sendToAll(jsonObject.toString());
+			// logging to logger
+			logMessage = "The onMessageReceived method is successfully implemented";
+			logger.log(ModuleID.INFRASTRUCTURE, LogLevel.SUCCESS, logMessage);
+		}
+		else if (meta.equals("userExit")) {
+			// logging to logger
+			logMessage = "User Exit case";
+			logger.log(ModuleID.INFRASTRUCTURE, LogLevel.INFO, logMessage);
+			try {
+				userName = jsonObject.getString("username");
+			} catch(Exception e) {
+				// logging to logger
+				logMessage = "the value of username key in argument message is not a String";
+				logger.log(ModuleID.INFRASTRUCTURE, LogLevel.ERROR, logMessage);
+				return;
+			}
+			imageMap.remove(userName);
+			// logging to logger
+			logMessage = "The message is now being broadcasted to all clients";
+			logger.log(ModuleID.INFRASTRUCTURE, LogLevel.INFO, logMessage);
+			sendToAll(message);
+			// logging to logger
+			logMessage = "The onMessageReceived method is successfully implemented";
+			logger.log(ModuleID.INFRASTRUCTURE, LogLevel.SUCCESS, logMessage);
+		}
+		else {
+			// logging to logger
+			logMessage = "The value of meta key of argument message is an invalid String";
+			logger.log(ModuleID.INFRASTRUCTURE, LogLevel.ERROR, logMessage);
+			return;
+		}
+	}
+	
+	/**
+	 * This method is used to send message to each active client of the board. It first gets the latest
+	 * userName to ipAddress map from processing and uses the ipAddress to send message to all clients. 
+	 * @param message - This String argument is going to be sent to all active clients.
+	 */
+	private void sendToAll(String message) {
+		// logging to logger
+		logMessage = "sendToAll method of ContentServer class is going to be executed";
+		logger.log(ModuleID.INFRASTRUCTURE, LogLevel.INFO, logMessage);
+		// Obtaining the userName to ipAddress map from processing
+		userIpMap = (HashMap<Username, IpAddress>) clientIP.getClientIP();
+		for (Username name: userIpMap.keySet()) {
+			userName = name.toString();
+			ipAddress = userIpMap.get(name);
+			//sending message to ipAdress via networking

```

Comment:
Shouldn't the parameter be userName?(Since that is the extracted variable)

Reply:
No, name is of type Username, userName is of type String